tests/enterprise-contract/contract.go

```diff
@@ -0,0 +1,146 @@
+package enterprisecontract
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/devfile/library/pkg/util"
+	ecp "github.com/enterprise-contract/enterprise-contract-controller/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/e2e-tests/pkg/constants"
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils/tekton"
+)
+
+var _ = framework.EnterpriseContractSuiteDescribe("Enterprise Contract E2E tests", Label("enterprise-contract", "HACBS"), func() {
+
+	defer GinkgoRecover()
+	var fwk *framework.Framework
+	var err error
+	var namespace string
+	var kubeController tekton.KubeController
+	var policySource []ecp.Source
+	var imageWithDigest string
+	var pipelineRunTimeout int
+	var generator tekton.VerifyEnterpriseContract
+	var verifyECTaskBundle string
+	publicSecretName := "cosign-public-key"
+
+	BeforeAll(func() {
+		fwk, err = framework.NewFramework(constants.TEKTON_CHAINS_E2E_USER)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(fwk.UserNamespace).NotTo(BeNil(), "failed to create sandbox user")
+		namespace = fwk.UserNamespace
+		kubeController = tekton.KubeController{
+			Commonctrl: *fwk.AsKubeAdmin.CommonController,
+			Tektonctrl: *fwk.AsKubeAdmin.TektonController,
+			Namespace:  namespace,
+		}
+		publicKey, err := kubeController.GetTektonChainsPublicKey()
+		Expect(err).ToNot(HaveOccurred())
+		GinkgoWriter.Printf("Copy public key from %s/signing-secrets to a new secret\n", constants.TEKTON_CHAINS_NS)
+		Expect(kubeController.CreateOrUpdateSigningSecret(
+			publicKey, publicSecretName, namespace)).To(Succeed())
+
+		defaultEcp, err := kubeController.GetEnterpriseContractPolicy("default", "enterprise-contract-service")
+		Expect(err).NotTo(HaveOccurred())
+		policySource = defaultEcp.Spec.Sources
+
+		cm, err := kubeController.Commonctrl.GetConfigMap("ec-defaults", "enterprise-contract-service")
+		Expect(err).ToNot(HaveOccurred())
+		verifyECTaskBundle = cm.Data["verify_ec_task_bundle"]
+		Expect(verifyECTaskBundle).ToNot(BeEmpty())
+		GinkgoWriter.Printf("Using verify EC task bundle: %s\n", verifyECTaskBundle)
+
+	})
+
+	Context("Release Policy", func() {
+		BeforeAll(func() {
+			generator = tekton.VerifyEnterpriseContract{
+				Bundle:              verifyECTaskBundle,
+				Image:               imageWithDigest,
```

Comment:
When is `imageWithDigest` not an empty string?

Reply:
initialized imageWithDigest with latest golden image above.